File: iCinema.Application/DTOs/Reports/ReportRequestDto.cs
using System.ComponentModel.DataAnnotations;

namespace iCinema.Application.DTOs.Reports;

public class ReportRequestDto
{
    [Required]
    public ReportType ReportType { get; set; }
    
    [Required]
    public DateTime DateFrom { get; set; }
    
    [Required]
    public DateTime DateTo { get; set; }
}

public enum ReportType
{
    MovieReservations,
    MovieSales,
    HallReservations,
    CinemaReservations
}

File: iCinema.UI/icinema_desktop/lib/features/reports/data/report_request_dto.dart
import '../domain/report_type.dart';

class ReportRequestDto {
  final int reportType;
  final DateTime dateFrom;
  final DateTime dateTo;

  ReportRequestDto({
    required ReportType reportTypeEnum,
    required this.dateFrom,
    required this.dateTo,
  }) : reportType = reportTypeEnum.index;

  Map<String, dynamic> toJson() {
    return {
      'reportType': reportType,
      'dateFrom': dateFrom.toIso8601String(),
      'dateTo': dateTo.toIso8601String(),
    };
  }
}
